.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,77 @@
+name: CI
+
+# on:
+#   push:
+#     branches-ignore:
+#       - 'main'
+#       - '_bors*'
+
+on:
+  workflow_call:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  selection:
+    runs-on: hetzner-pm
+    container:
+      image: xaynetci/yellow:9135de0
+    outputs:
+      dart: ${{ steps.filter.outputs.dart }}
+      rust: ${{ steps.filter.outputs.rust }}
+      flutter: ${{ steps.filter.outputs.flutter }}
+    timeout-minutes: 5
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # v2.4.0
+
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721
+        id: filter
+        with:
+          filters: |
+            rust: &rust
+              - 'discovery_engine_core/**'
+              - '.github/workflows/ci*'
+              - '.github/actions/setup-job-docker/*'
+              - '.github/actions/setup-job/*'
+              - 'justfile'
+            dart: &dart
+              - *rust
+              - 'discovery_engine/**'
+            flutter:
+              - *rust
+              - *dart
+              - 'discovery_engine_flutter/**'
+
+  rust-checks:
+    needs: selection
+    if: ${{ needs.selection.outputs.rust  == 'true' }}
+    uses: ./.github/workflows/_reusable.rust.yml
```

Comment:
❔does this work now? I thought reusable workflow cannot be nested

Reply:
`pull-request` is a top-level workflow, so this would not qualify as nested, I think 🤔

Reply:
but `workflow_call` defines a resuable workflow. maybe `workflow_dispatch` was meant?

Reply:
Ah, I think that's just a misunderstanding. The `pull-request.yml` is what was previously the `ci.yml`, but I threw in the `workflow_call` so that it doesn't run automatically while I'm testing the full bors CI run :) So that `workflow_call` would not remain there.